xonsh/completers/path.py

```diff
@@ -151,6 +151,71 @@ def _quote_paths(paths, start, end):
     return out
 
 
+def _joinpath(path):
+    # convert our tuple representation back into a string representing a path
+    if path is None:
+        return ''
+    elif len(path) == 0:
+        return ''
+    elif path == ('',):
+        return get_sep()
+    elif path[0] == '':
+        return get_sep() + _normpath(os.path.join(*path))
+    else:
+        return _normpath(os.path.join(*path))
+
+
+def _splitpath(path):
+    # convert a path into an intermediate tuple representation
+    # if this tuple starts with '', it means that the path was an absolute path
+    path = _normpath(path)
+    if path.startswith(get_sep()):
```

Comment:
I did a little bit of testing on Windows and it seemed to work.  But you're definitely more expert there than I am.  Is there something that isn't working?